### PR TITLE
Feature/fuji 3523b

### PR DESCRIPTION
**Pull request by mrapptambour**

### CODE REVIEW

File: rets-client/dev/src/java/org/realtors/rets/client/CommonsHttpClient.java [hunk 1]
```diff
@@ -92,6 +92,11 @@ public void unsetUserCredentials(){
 		this.httpClient.getCredentialsProvider().clear();
 	}
 
+	@Override public String getCookieValue(String key) {
+		return this.getCookies().get(key);
```
Comment (mrapptambour) [inline rets-client/dev/src/java/org/realtors/rets/client/CommonsHttpClient.java]: @mmastroianni how does this differ from the non-override?

File: rets-client/dev/src/java/org/realtors/rets/client/RetsSession.java [hunk 2]
```diff
@@ -328,8 +339,11 @@ public SearchResult search(SearchRequest req) throws RetsException {
 	 * @exception RetsException if an error occurs
 	 */
 	public void search(SearchRequest req, SearchResultCollector collector) throws RetsException {
-		String sessionCookie = generateSessionCookie(this.sessionId);
```
Comment (mrapptambour) [inline rets-client/dev/src/java/org/realtors/rets/client/RetsSession.java]: @mmastroianni can the check to this.transport live inside generateSessionCookie?

File: rets-client/dev/src/java/org/realtors/rets/client/RetsSession.java [hunk 3]
```diff
@@ -340,8 +354,11 @@ public void search(SearchRequest req, SearchResultCollector collector) throws Re
 	 * @param processor the result object that will process the data
 	 */
 	public SearchResultSet search(SearchRequest req, SearchResultProcessor processor) throws RetsException {
-		String sessionCookie = generateSessionCookie(sessionId);
```
Comment (mrapptambour) [inline rets-client/dev/src/java/org/realtors/rets/client/RetsSession.java]: @mmastroianni same thing here. Does generateSessionCookie not have access to this.transport or something?

File: rets-client/dev/src/java/org/realtors/rets/client/CommonsHttpClient.java (hunk 1 above)
Comment (mmastroianni) [inline rets-client/dev/src/java/org/realtors/rets/client/CommonsHttpClient.java]: Just that someone with access to the client via a base class reference can access the getCookieValue method, which shouldn't be specific to apache commons based clients

**Pull request merged**